testing-overhead/src/test/java/io/opentelemetry/LatestAgentSnapshotResolver.java

```diff
@@ -0,0 +1,83 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+package io.opentelemetry;
+
+import static org.joox.JOOX.$;
+
+import java.io.FileOutputStream;
+import java.io.IOException;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.Optional;
+import okhttp3.OkHttpClient;
+import okhttp3.Request;
+import okhttp3.Response;
+import okhttp3.ResponseBody;
+import org.joox.Match;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.w3c.dom.Document;
+
+public class LatestAgentSnapshotResolver {
+
+  private static final Logger logger = LoggerFactory.getLogger(LatestAgentSnapshotResolver.class);
+
+  final static String BASE_URL = "https://oss.sonatype.org/content/repositories/snapshots/io/opentelemetry/javaagent/opentelemetry-javaagent";
+  final static String LATEST_SNAPSHOT_META = BASE_URL + "/maven-metadata.xml";
+
+  Optional<Path> resolve() throws IOException {
+    String version = fetchLatestSnapshotVersion();
+    logger.info("Latest snapshot version is {}", version);
+    String latestFilename = fetchLatestFilename(version);
+    String url = BASE_URL +  "/" + version + "/" + latestFilename;
+    byte[] jarBytes = fetchBodyBytesFrom(url);
+    Path path = Paths.get(".", "opentelemetry-javaagent-SNAPSHOT-all.jar");
+    FileOutputStream out = new FileOutputStream(path.toFile());
+    out.write(jarBytes);
+    return Optional.of(path);
+  }
+
+  private String fetchLatestFilename(String version) throws IOException {
```

Comment:
Please take a look at Eclipse Aether and how we use it `io.opentelemetry.instrumentation.muzzle-check.gradle.kts`. It may be easier/more robust than writing this code manually.

Reply:
Thanks, I'll take a look. Writing this code definitely felt bad.

Reply:
Oof, [aether is archived](https://projects.eclipse.org/projects/technology.aether). Are we sure that we'd want to add another dependency on an archived (dead) project?  Last release was 6.5 years ago.  😢

Reply:
:( Probably not :(

Reply:
Well, if we find a better contemporary way of resolving latest/snapshot versions of jar files we can overhaul the resolver at a future time. At least it is abstracted.